src/ray/raylet/local_task_manager.cc

```diff
@@ -387,6 +387,16 @@ void LocalTaskManager::DispatchScheduledTasksToWorkers() {
                 const std::shared_ptr<WorkerInterface> worker,
                 PopWorkerStatus status,
                 const std::string &runtime_env_setup_error_message) -> bool {
+              // TODO(hjiang): After getting the ready-to-use worker and task id, we're
```

Comment:
Wait a minute... The `running_tasks` is inserted with task ID, *before* we call `PopWorker`. This means if you change running_tasks to a `map { task ID -> cgroup raii }`, the value must be nullptr at first, and is only set until PopWorker callback is called. I'm OK with this, but, @jjyao do you feel putting the cgroup raii (to remove proc from cgroup and delete cgroup on dtor) in LocalTaskManager's running_tasks is the best place? Is there a even more precise data structure somewhere in raylet to model a running task attempt?

Reply:
> the value must be nullptr at first, and is only set until PopWorker callback is called

Yeah that's my plan

Reply:
A nullptr is needed anyway I think, for example, we need to handle cases where cgroup prerequisite is not met.

Reply:
@jjyao Do you have any thoughts?